OpenRA.Mods.Common/Traits/Harvester.cs

```diff
@@ -102,17 +98,35 @@ public class HarvesterInfo : ConditionalTraitInfo, Requires<MobileInfo>
 		public readonly string HarvestCursor = "harvest";
 
 		public override object Create(ActorInitializer init) { return new Harvester(init.Self, this); }
+
+		void IRulesetLoaded<ActorInfo>.RulesetLoaded(Ruleset rules, ActorInfo info)
+		{
+			var resourceTypes = Resources.ToList();
+
+			if (resourceTypes.Count == 0)
+				throw new YamlException($"Harvester.{nameof(Resources)} is empty.");
+
+			foreach (var sr in info.TraitInfos<IStoresResourcesInfo>())
+			{
+				foreach (var type in sr.ResourceTypes)
+				{
+					if (Resources.Contains(type) && resourceTypes.Contains(type))
+						resourceTypes.Remove(type);
+				}
+			}
```

Comment:
```suggestion
			foreach (var sr in info.TraitInfos<IStoresResourcesInfo>())
				foreach (var type in sr.ResourceTypes)
					if (Resources.Contains(type))
						resourceTypes.Remove(type);
```

Reply:
Or something like
```suggestion
			foreach (var type in info.TraitInfos<IStoresResourcesInfo>().SelectMany(sr => sr.ResourceTypes))
				if (Resources.Contains(type))
					resourceTypes.Remove(type);
```
if you prefer that.

Reply:
made it fully use Linq